### PR TITLE
x1 updates

### PR DESCRIPTION
**Pull request by aubergemediale**

### CODE REVIEW

File: src/NBitcoin/TransactionBuilder.cs [hunk 1]
```diff
@@ -72,12 +72,12 @@ public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
             if (target.CompareTo(zero) == 0)
                 return result;
 
-            var orderedCoinGroups = coins.GroupBy(c => this.GroupByScriptPubKey ? c.TxOut.ScriptPubKey : new Key().ScriptPubKey)
+            var orderedCoinGroups = coins.ToArray().GroupBy(c => this.GroupByScriptPubKey ? c.TxOut.ScriptPubKey : new Key().ScriptPubKey)
```
Comment (dangershony) [inline src/NBitcoin/TransactionBuilder.cs]: why enumerate the collection?

File: src/Obsidian.Networks.ObsidianX/ObsidianXMain.cs [hunk 2]
```diff
@@ -33,9 +33,9 @@ public ObsidianXMain()
             this.DefaultBanTimeSeconds = 16000; // 500 (MaxReorg) * 64 (TargetSpacing) / 2 = 4 hours, 26 minutes and 40 seconds
 
             this.MaxTipAge = 2 * 60 * 60; // (120 minutes)
-            this.MinTxFee = 100;
-            this.FallbackFee = 500;
-            this.MinRelayTxFee = 100;
+            this.MinTxFee = Money.Coins(0.001m).Satoshi;
```
Comment (dangershony) [inline src/Obsidian.Networks.ObsidianX/ObsidianXMain.cs]: I would recommend to separate changes between features, a PR for X1 changes and a fee PR (unless they are strongly linked)

Comment (dangershony) [reply src/Obsidian.Networks.ObsidianX/ObsidianXMain.cs]: Ah sorry ignore this, I realize you just merged the other PRs

**Comment by aubergemediale** [comment]
I will not merge this PR without decision on the two other PRs. But I run 2 nodes atm with the two other PRs active (works fine, no issues). SO you can consider the other PRs as already tested, they do not break anything.

**Comment by aubergemediale** [comment]
none of these changes will break the Stratis wallet, it works all correctly! I have tested with the TransactionBuilder which the Stratis wallet uses and hand-made transactions. It all works fine!

**Comment by aubergemediale** [comment]
http://explorer.obsidianproject.org/fetchblock/1277

Checkout this wonderful block, look at the size and the transaction :-)

It has 1000 outputs and 11 inputs. I did the tx and mined the block. Got 5.032655 ODX as reward. That's not to high a fee, or is it?

**Comment by dangershony** [comment]
That seems fine to me the block size is 32kb (I think the explorer measurement is incorrect it should be bytes)
This is an ok fee

File: src/NBitcoin/TransactionBuilder.cs (hunk 1 above)
Comment (aubergemediale) [reply src/NBitcoin/TransactionBuilder.cs]: This change is reverted!

Reason was testing if that fixes the StackoverflowException (the collection get can enumerated several times). It does not fix it.

**Comment by aubergemediale** [comment]
The current state now matches this release:
https://github.com/obsidianproject/Obsidian-StratisNode/releases/tag/v0.4-beta
If there are no accidentally affected files outside X1, it would be a good time to merge now.

**Pull request merged**